bot/exts/events/advent_of_code/_cog.py

```diff
@@ -96,7 +96,9 @@ async def completionist_task(self) -> None:
                 # Only give the role to people who have completed all 50 stars
                 continue
 
-            member_id = aoc_name_to_member_id.get(member_aoc_info["name"], None)
+            aoc_name = member_aoc_info["name"] or f"Anonymous #{member_aoc_info['id']}"
```

Comment:
```suggestion
            aoc_name = member_aoc_info.get("name") or f"Anonymous #{member_aoc_info['id']}"
```

Why isn't the name missing from `member_aoc_info` or was that an incorrect assumption?

Reply:
I should have been more specific, the key is present, but is empty.